model/src/main/kotlin/Vulnerability.kt

```diff
@@ -45,30 +45,52 @@ data class Vulnerability(
     val url: URL? = null
 ) {
     /**
-     * The rating attaches human-readable semantics to the score number, see
-     * https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale.
+     * The rating attaches human-readable semantics to the score number according to CVSS version 2, see
+     * https://www.first.org/cvss/v2/guide.
      */
-    enum class Rating(private val upperBound: Float) {
+    enum class Cvss2Rating(private val upperBound: Float) {
+        LOW(4.0f),
+        MEDIUM(7.0f),
+        HIGH(10.0f);
+
+        companion object {
+            /**
+             * Get the [Cvss2Rating] from a [score], or null if the [score] does not map to any [Cvss2Rating].
+             */
+            fun fromScore(score: Float): Cvss2Rating? =
+                when {
+                    score < 0.0f || score > HIGH.upperBound -> null
+                    score < LOW.upperBound -> LOW
+                    score < MEDIUM.upperBound -> MEDIUM
+                    score <= HIGH.upperBound -> HIGH
+                    else -> null
+                }
+        }
+    }
+
+    /**
+     * The rating attaches human-readable semantics to the score number according to CVSS version 3, see
+     * https://www.first.org/cvss/v3.0/specification-document#Qualitative-Severity-Rating-Scale.
+     */
+    enum class Cvss3Rating(private val upperBound: Float) {
         NONE(0.0f),
         LOW(4.0f),
         MEDIUM(7.0f),
         HIGH(9.0f),
         CRITICAL(10.0f);
 
         companion object {
-            private val bounds = enumValues<Rating>().asIterable().zipWithNext()
+            private val bounds = enumValues<Cvss3Rating>().asIterable().zipWithNext()
             private val limits = listOf(bounds.first().first, bounds.last().second)
 
             /**
-             * Get the [Rating] from a [score], or null if the [score] does not map to any [Rating].
+             * Get the [Cvss3Rating] from a [score], or null if the [score] does not map to any [Cvss3Rating].
              */
-            fun fromCvssScore(score: Float): Rating? {
+            fun fromScore(score: Float): Cvss3Rating? {
```

Comment:
Both `Cvss2Rating` and `Cvss3Rating` provide `fromScore()` functions, which are semantically equivalent, but use completely different algorithms to calculate the score. This is a bit confusing. Should the implementations be aligned?

Reply:
Hmm, I'm not sure I get what you mean here. The implementations *have* to be different as the specs define the mappings differently for v2 and v3. So how could the implementations be aligned?

Reply:
What I meant is that the implementations could be structurally similar. Currently, one uses a `when` statement while the other iterates over the enum values.

Reply:
I actually liked the fact to show off two different ways how these could be implemented, and I usually prefer generic over hard-coded solutions, but in this case I can see your point, and I've added a new commit to align the implementations.